maasglobal-schema-generator-ajv/src/main.ts

```diff
@@ -44,26 +44,35 @@ export const generateClient: CodeGenerator = (sourceDir) => {
   return `import * as mjsv from 'maasglobal-json-schema-validator';
 
 let validator: mjsv.Validator|undefined;
+let validatorNonStrict: mjsv.Validator|undefined;
 
-export function init(): mjsv.Validator {
+export function init(): [mjsv.Validator, mjsv.Validator] {
   validator = mjsv.validator([
 ${registries.join(',\n')}
   ]);
+  validatorNonStrict = mjsv.validatorNonStrict([
+${registries.join(',\n')}
+  ]);
 
-  return validator;
+  return [validator, validatorNonStrict];
 }
 
-export function cached(): mjsv.Validator {
-  if (typeof validator !== 'undefined') {
-    return validator
+export function cached(): [mjsv.Validator, mjsv.Validator] {
+  if (typeof validator !== 'undefined' && typeof validatorNonStrict !== 'undefined') {
+    return [validator, validatorNonStrict];
   }
   return init();
 }
 
 export function validate(schema: mjsv.Schema|mjsv.SchemaURI, obj: unknown): unknown {
-  const instance = cached()
+  const [instance, _] = cached()
   return instance.validate(schema, obj);
 }
+
+export function validateNonStrict(schema: mjsv.Schema|mjsv.SchemaURI, obj: unknown): unknown {
```

Comment:
I also wonder if adding a deprecated thing to the generator output makes sense. If it is truly deprecated you might want to use it directly in backend without adding code generation support for the deprecated mechanism.

Reply:
Removed.